projects/js-packages/publicize-components/src/components/form/index.tsx

```diff
@@ -25,8 +29,16 @@ import { SharePostForm } from './share-post-form';
  * @return {object} - Publicize form component.
  */
 export default function PublicizeForm() {
-	const { hasConnections, hasEnabledConnections } = useSocialMediaConnections();
+	const { hasConnections, hasEnabledConnections, connections } = useSocialMediaConnections();
 	const { isPublicizeEnabled, isPublicizeDisabledBySitePlan } = usePublicizeConfig();
+	const { attachedMedia } = useAttachedMedia();
+	const featuredImageId = useFeaturedImage();
+
+	const mediaId = attachedMedia[ 0 ]?.id || featuredImageId;
+	const { isConvertible } = useMediaRestrictions( connections, useMediaDetails( mediaId )[ 0 ] );
+
+	const showSharePostForm =
+		isPublicizeEnabled && ( hasEnabledConnections || attachedMedia.length > 0 || ! isConvertible );
```

Comment:
`isConvertible` is `false` if there is no media attached. Thus `! isConvertible` becomes `true`.

That means, the `SharePostForm` is visible by default, which is not the case on trunk.

<img width="305" alt="image" src="https://github.com/user-attachments/assets/4b13f9de-aab0-4758-8964-54c0b82fcfd7">

Reply:
So we need the full logic as before

Reply:
May be we don't need to use `isConvertible` here?

Reply:
I think we do because otherwise we would show the modal even if the media is invalid but convertible. For example the featured image is too big, but can be converted, in that case I think we do not need it 🤔 This logic starts to get fiddly

Reply:
Yeah, this logic is going to be difficult to untangle in future.

May be we can add explanatory comments to that code.